src/main/java/hudson/tasks/junit/JUnitResultArchiver.java

```diff
@@ -282,7 +285,17 @@ public static TestResultSummary parseAndSummarize(@Nonnull JUnitTask task, Pipel
             }
 
             if (!task.isSkipPublishingChecks()) {
-                new JUnitChecksPublisher(build, task.getChecksName(), result, summary).publishChecks(listener);
+                // If we haven't been provided with a checks name, and we have pipeline test details, set the checks name
+                // to be a ' / '-joined string of the enclosing blocks names. If all of that ends up being empty or null,
+                // set a default of 'Tests'
+                String checksName = task.getChecksName();
```

Comment:
shouldn't we still include tests in the check ?

![image](https://user-images.githubusercontent.com/21194782/100987645-ae1c3880-3546-11eb-8d42-8185e6c9b09b.png)

Reply:
Hmm, yes. For context I'm working mostly on python/js environments, where our stages are dedicated to test generation...

```groovy
stage('JS Tests') {
    sh 'yarn test'
    junit 'js_tests.xml'
}

stage('Python Tests') {
    sh 'pytest tests/'
    junit 'python_tests.xml'
}
```

...as opposed to maven where tests are a bit more of a 'side effect' of the build

```groovy
stage('Build') {
    sh 'mvn install'
    junit 'results.xml'
    archiveArticfcats(...)
    recordIssues(...)
    // etc...
}
```

So we'd end up with `JS Tests / Tests` etc. OTOH, I'm already setting a custom name (and will be using `withChecks` when it's ready) so I think meeting the 'mvn' use case possibly makes more sense?

Reply:
Yeah I think so

Reply:
I made the change locally,

```
+
+    private static final String DEFAULT_CHECKS_NAME = "Tests";

     /**
      * {@link FileSet} "includes" string, like "foo/bar/*.xml"
@@ -290,10 +294,12 @@ public class JUnitResultArchiver extends Recorder implements SimpleBuildStep, JU
                 // set a default of 'Tests'
                 String checksName = task.getChecksName();
                 if (checksName == null && pipelineTestDetails != null) {
-                    checksName = StringUtils.join(new ReverseListIterator(pipelineTestDetails.getEnclosingBlockNames()), " / ");
+                    List<String> checksComponents = new ArrayList<>(pipelineTestDetails.getEnclosingBlockNames());
+                    checksComponents.add(DEFAULT_CHECKS_NAME);
+                    checksName = StringUtils.join(new ReverseListIterator(checksComponents), " / ");
                 }
                 if (Util.fixEmpty(checksName) == null) {
-                    checksName = "Tests";
+                    checksName = DEFAULT_CHECKS_NAME;
                 }
```

Requires tests changes but that gives what I expected

Reply:
OK, will get that done

Reply:
Ugh, sorry, you meant `Tests` at the beginning...